File: 그리디/QueueReconstructionByHeight/reconstruct_queue.py
class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        sorted_people = sorted(people, key = lambda person: (-person[0], person[1]))
        answer = []
        for person in sorted_people:
            answer.insert(person[1], person)
        return answer

print(Solution().reconstructQueue([[7,0], [4,4], [7,1], [5,0], [6,1], [5,2]]))

File: 그리디/queue_reconstruction_by_height/sehoongim/reconstruct_queue.rb
def reconstruct_queue(people)
  people.sort!{|a,b| a[0] == b[0] ? a[1] - b[1] : b[0] - a[0]}
  answer = []
  people.each do |person|
    answer.insert(person[1], person)
  end
  answer
end

p reconstruct_queue [[7,0], [4,4], [7,1], [5,0], [6,1], [5,2]]